File: bench/interp/bench_method.js
class Vec {
    constructor(x, y) { this.x = x; this.y = y; }
    AddX(d) { this.x = this.x + d; }
    Dot() { return this.x * this.y; }
}
const v = new Vec(1, 3);
const n = 6000000;
const t0 = Date.now();
let total = 0;
for (let i = 0; i < n; i++) { v.AddX(1); total += v.Dot(); }
console.log("result: " + total);
console.log("elapsed: " + (Date.now() - t0) + " ms");

File: bench/interp/bench_method.py
import time
class Vec:
    __slots__ = ('x', 'y')
    def __init__(self, x, y): self.x = x; self.y = y
    def AddX(self, d): self.x = self.x + d
    def Dot(self): return self.x * self.y
v = Vec(1, 3)
n = 6000000
t0 = time.perf_counter()
total = 0
for _ in range(n):
    v.AddX(1)
    total += v.Dot()
print("result: %d" % total)
print("elapsed: %d ms" % int((time.perf_counter()-t0)*1000))
